src/sagas/index.js

```diff
@@ -343,19 +390,39 @@ export function* startProgress ({ vmId, poolId, name }) {
 }
 
 export function* stopProgress ({ vmId, poolId, name, result }) {
-  if (result && result.status === 'complete') {
-    vmId = vmId || result.vm.id
-    // do not call 'end of in progress' if successful,
-    // since UI will be updated by refresh
-    yield delay(5 * 1000)
-    yield getSingleInstance({ vmId, poolId })
-
-    yield delay(30 * 1000)
-    yield getSingleInstance({ vmId, poolId })
+  if (result?.status === 'complete' && result?.job?.id) {
+    // If the result is complete, refresh the entity to grab the current status
+    if (vmId) {
+      yield put(getSingleVm({ vmId, shallowFetch: true }))
```

Comment:
Because at least half of the VM actions seem to be done between the time they're requrested and this first check w/o delay.  With the shallow fetch, the overhead for the fetch is very low.